File: c/examples/servo_control_modes.c
/******************************************************************************
     * File: servo_control_modes.c
     * Description: Пример работы с сервоприводом . Работа в различных ре-имах управления дигателем.
******************************************************************************/
#include "Servo.h"
#include "bus_handler.h"

//Адрес сервопривода. По умолчанию 10.
int servo_addr = 10;

int main()
{

  //Инициализация последовательного порта для работы с сервоприводом. По умолчанию /dev/RS_485
  if (init_port("/dev/RS_485",false) < 0)
  {
    return -1;
  }

  //Инициализация сервопривода
  init_servo(servo_addr);

  //Стандартный ре-им работы сервопривода
  set_PID_mode(servo_addr,NORMAL);

  //Включение питания обмоток двигателя
  set_torque(servo_addr,true);

  //Перемещение на позицию 4000
  set_point(servo_addr,4000);

  sleep(4);

  //Выключение питания обмоток двигателя
  set_torque(servo_addr,false);

  //Ре-им ПИД по скорости
  set_PID_mode(servo_addr,SPEED);

  set_torque(servo_addr,true);

   //Скорость вращения вала 10 об/сек
  set_point(servo_addr,10);

  sleep(4);  

  set_torque(servo_addr,false);

  //Прямой ре-им работы. Без ПИД регуляторов
  set_PID_mode(servo_addr,PWM);
  set_torque(servo_addr,true);

  //Высталение ШИМ 750 из 1000. Сква-ность 75%
  set_point(servo_addr,750);
  sleep(4);  
  set_torque(servo_addr,false);

  //Включение стадартного ре-има работы
  set_PID_mode(servo_addr,NORMAL);
 
  return 0; 
}

File: python/examples/servo_control_modes.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import Servo
import time
"""
servo_mode_to.py

Пример работы сервопривода в разных режимах.

* С каскадом ПИД регуляторов по сокрости и положение.

* Только с ПИД регулятором по скорости.

* Без регуляторов отпрвка ШИМ сигнала напрямую.

"""

#Порт шины данных. По умолчпнию /dev/RS_485
port        = '/dev/RS_485'

#Адрес сервопривода. По умолчанию 10
servo_id    = 10

#Инициализация сервопривода
servo = Servo.Servo(port,servo_id,debug = False)


if __name__ == '__main__':

    #С каскадом ПИД регуляторов по сокрости и положение.
    print("NORMAL mode")

    servo.set_PID_Mode("NORMAL")

    #Включение питания обмоток мотора
    servo.set_torque(1)

    #Движение в позицию 4000        
    servo.set_point(4000)
     
    time.sleep(4)


    #Только с ПИД регулятором по скорости.
    print("SPEED mode")
    servo.set_torque(0)                

    servo.set_PID_Mode("SPEED")

    servo.set_torque(1)

    #Установка оборотво вала в минуту
    servo.set_point(10)

    time.sleep(4)

    servo.set_torque(0)    


    #Без регуляторов, отправка ШИМ сигнала напрямую.
    print("PWM mode")
    servo.set_torque(0)                

    servo.set_PID_Mode("PWM")

    servo.set_torque(1)

    #Установка скважности ШИМ сигнала. (0-1000)
    servo.set_point(750)

    time.sleep(4)

    servo.set_torque(0)      


    #Выставление обычного режима работы           
    servo.set_PID_Mode("NORMAL")   